ibis/backends/tests/test_set_ops.py

```diff
@@ -139,6 +140,52 @@ def test_difference(backend, alltypes, df, distinct):
 
 
 @pytest.mark.parametrize("method", ["intersect", "difference", "union"])
-def test_empty_set_op(backend, alltypes, method):
+def test_empty_set_op(alltypes, method):
     with pytest.raises(com.IbisTypeError, match="requires a table or tables"):
         getattr(alltypes, method)()
+
+
+@pytest.mark.parametrize("distinct", [True, False])
+@pytest.mark.notimpl(["dask", "pandas"], raises=com.UnboundExpressionError)
+@pytest.mark.notyet(
+    ["clickhouse"],
+    raises=sg.errors.ParseError,
+    reason="sqlglot fails to parse UNION ALL/DISTINCT",
+)
+def test_top_level_union(con, distinct, monkeypatch):
+    monkeypatch.setattr(ibis.options, "default_backend", con)
+    monkeypatch.setattr(ibis.options, "interactive", True)
```

Comment:
Why is this new testing approach?

Reply:
What do you mean?

Reply:
I'm setting the default backend so that I can transparently test using `memtable`. I'm setting interactive mode since that is require to reproduce the failure.

Reply:
I can't see any issue referenced, based on the select builder changes it looks like a new feature, that's why I find the testing weird.

Reply:
I think I'm still not following :)

How is whether the thing is a new feature versus a fix related to how it's being tested?

Reply:
I'd expect to test like we usually do, execute the expression and compare it with the expected pandas result.

Reply:
Fixed.